src/type/validate.js

```diff
@@ -314,94 +317,106 @@ function validateFields(
   }
 }
 
-function validateObjectInterfaces(
+function validateInterfaces(
   context: SchemaValidationContext,
-  object: GraphQLObjectType,
+  type: GraphQLObjectType | GraphQLInterfaceType,
 ): void {
   const implementedTypeNames = Object.create(null);
-  for (const iface of object.getInterfaces()) {
+  for (const iface of type.getInterfaces()) {
     if (!isInterfaceType(iface)) {
       context.reportError(
-        `Type ${inspect(object)} must only implement Interface types, ` +
+        `Type ${inspect(type)} must only implement Interface types, ` +
           `it cannot implement ${inspect(iface)}.`,
-        getAllImplementsInterfaceNodes(object, iface),
+        getAllImplementsInterfaceNodes(type, iface),
       );
       continue;
     }
 
     if (implementedTypeNames[iface.name]) {
       context.reportError(
-        `Type ${object.name} can only implement ${iface.name} once.`,
-        getAllImplementsInterfaceNodes(object, iface),
+        `Type ${type.name} can only implement ${iface.name} once.`,
+        getAllImplementsInterfaceNodes(type, iface),
       );
       continue;
     }
     implementedTypeNames[iface.name] = true;
-    validateObjectImplementsInterface(context, object, iface);
+    validateTypeImplementsAncestors(context, type, iface);
+    validateTypeImplementsInterface(context, type, iface);
   }
 }
 
-function validateObjectImplementsInterface(
+function validateTypeImplementsInterface(
   context: SchemaValidationContext,
-  object: GraphQLObjectType,
+  type: GraphQLObjectType | GraphQLInterfaceType,
   iface: GraphQLInterfaceType,
 ): void {
-  const objectFieldMap = object.getFields();
-  const ifaceFieldMap = iface.getFields();
+  const implementingFieldMap = type.getFields();
+  const implementedFieldMap = iface.getFields();
 
   // Assert each interface field is implemented.
-  for (const [fieldName, ifaceField] of objectEntries(ifaceFieldMap)) {
-    const objectField = objectFieldMap[fieldName];
+  for (const [fieldName, implementedField] of objectEntries(
+    implementedFieldMap,
+  )) {
+    const implementingField = implementingFieldMap[fieldName];
 
-    // Assert interface field exists on object.
-    if (!objectField) {
+    // Assert interface field exists on implementing.
+    if (!implementingField) {
       context.reportError(
-        `Interface field ${iface.name}.${fieldName} expected but ${object.name} does not provide it.`,
-        [ifaceField.astNode, ...getAllNodes(object)],
+        `Interface field ${iface.name}.${fieldName} expected but ${type.name} does not provide it.`,
+        [implementedField.astNode, ...getAllNodes(type)],
       );
       continue;
     }
 
-    // Assert interface field type is satisfied by object field type, by being
+    // Assert interface field type is satisfied by implementing field type, by being
```

Comment:
After the change, this comment doesn't make sense `is satisfied by implementing field type` is different from `is satisfied by object field type`.
Arguments here is the same as in this thread:
https://github.com/graphql/graphql-js/pull/2084#discussion_r328241103
So can we revert to the previous state and have `typeField` and `ifaceField`?
and thus minimize changes in this function.